[WebAuthn] Create local authenticator get credentials by RP/credential id SPI

pascoej opened the pull request:
#### 605a46e559cb52dca2222117b79b4d0dca452cef
<pre>
[WebAuthn] Create local authenticator get credentials by RP/credential id SPI
<a href="https://bugs.webkit.org/show_bug.cgi?id=242355">https://bugs.webkit.org/show_bug.cgi?id=242355</a>
&lt;rdar://96461832&gt;

Reviewed by Brent Fulgham.

These SPI provide the ability to filter platform credentials by ID or RP ID
for internal needs.

* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
(getAllLocalAuthenticatorCredentialsImpl):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentials]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithAccessGroup:]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithRPID:]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithCredentialID:]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithRPIDAndAccessGroup:rpID:]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithCredentialIDAndAccessGroup:credentialID:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanelForTesting.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/_WKWebAuthenticationPanel.mm:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/252289@main">https://commits.webkit.org/252289@main</a>
</pre>

brentfulgham commented on Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
```diff
@@ -250,20 +250,25 @@ - (void)setDelegate:(id<_WKWebAuthenticationPanelDelegate>)delegate
         (__bridge id)kSecUseDataProtectionKeychain: @YES
-    };
+    }];
+    if (byRpId != nil)
```
I think we usually just write this as:
```
if (byRpId)
...
```

brentfulgham reviewed (commented):
Looks good, though I don't think we typically test for equality against 'nil'.

pascoej committed:
[WebAuthn] Create local authenticator get credentials by RP/credential id SPI
https://bugs.webkit.org/show_bug.cgi?id=242355
<rdar://96461832>

Reviewed by Brent Fulgham.

These SPI provide the ability to filter platform credentials by ID or RP ID
for internal needs.

* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
(getAllLocalAuthenticatorCredentialsImpl):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentials]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithAccessGroup:]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithRPID:]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithCredentialID:]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithRPIDAndAccessGroup:rpID:]):
(+[_WKWebAuthenticationPanel getAllLocalAuthenticatorCredentialsWithCredentialIDAndAccessGroup:credentialID:]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanelForTesting.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/_WKWebAuthenticationPanel.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/252289@main
